### PR TITLE
Allow Cross-Compile with latest Ubuntu and Raspbian versions

### PR DESCRIPTION
**Pull request by Lord-Grey**
Updates allowing cross-compiling for Raspberry PI using latest Ubuntu and Raspbian versions.
Tested on Ubuntu 18.10, 19.04, Raspbian Stretch and RPi 3B+

### CODE REVIEW

File: dependencies/CMakeLists.txt [hunk 1]
```diff
@@ -41,8 +41,14 @@ else ()
 	set(FLATBUFFERS_INCLUDE_DIRS ${FLATBUFFERS_INCLUDE_DIRS} PARENT_SCOPE)
 
 	# define the flatc executable at the parent scope
-	get_property(FLATBUFFERS_FLATC_EXECUTABLE TARGET flatc PROPERTY LOCATION)
+	#get_property(FLATBUFFERS_FLATC_EXECUTABLE TARGET flatc PROPERTY LOCATION)
```
Comment (Paulchen-Panther) [inline dependencies/CMakeLists.txt]: Please uncomment this for native compiling.

File: dependencies/CMakeLists.txt [hunk 2]
```diff
@@ -41,8 +41,14 @@ else ()
...
+
+	#Workaround, set flatc comiplier directory hard, as cmake definitions of flatc do not cater for crosscompile correctly.
+	#Includ of flatc_export.cmake detects that flatc target is defined aand returns before using the definitions written by export
+	set ( FLATBUFFERS_FLATC_EXECUTABLE "${CMAKE_BINARY_DIR}/../build-x86x64/bin/flatc")
```
Comment (Paulchen-Panther) [inline dependencies/CMakeLists.txt]: In case of cross compiling: check the variable "CMAKE_CROSSCOMPILING" with an “if” statement and set the flatc

**Comment by Paulchen-Panther** [comment]
Thanks for your work. Could you please set the schema compiler individually, eg. in the form of an if instruction. Many thanks in advance.
Greetings, Paulchen.

**Comment by Lord-Grey** [comment]
Updated as suggested.

**Comment by Paulchen-Panther** [comment]
Thank you

**Comment by RickDB** [comment]
Just let me know if it's ready to merge and will add it in :)

**Comment by Lord-Grey** [comment]
Ready to merge

**Comment by RickDB** [comment]
One new conflict due to our Travis fix which excludes some ARM platforms due to time restraints on the Travis end.

 @Lord-Grey could you update the CMakeLists.txt?

**Comment by Lord-Grey** [comment]
I merged the CMakeLists.txt files.

**Comment by RickDB** [comment]
Thanks!

**Pull request merged**